ytsync/ytsync.go

```diff
@@ -61,16 +61,27 @@ type Sync struct {
 	Refill                  int
 	Manager                 *SyncManager
 
-	daemon         *jsonrpc.Client
-	claimAddress   string
-	videoDirectory string
-	db             *redisdb.DB
-
-	grp *stop.Group
+	daemon          *jsonrpc.Client
+	claimAddress    string
+	videoDirectory  string
+	db              *redisdb.DB
+	syncedVideos    map[string]syncedVideo
+	syncedVideosMux *sync.Mutex
+	grp             *stop.Group
+	lbryChannelID   string
+
+	walletMux *sync.Mutex
+	queue     chan video
+}
 
-	mux   sync.Mutex
-	wg    sync.WaitGroup
-	queue chan video
+func (s *Sync) AppendSyncedVideo(videoID string, published bool, failureReason string) {
+	s.syncedVideosMux.Lock()
+	defer s.syncedVideosMux.Unlock()
```

Comment:
I don't understand why we have a mutex lock here. Why would this get called twice in different go routines on the same sync instance?

Reply:
videos end concurrently, that's why